### PR TITLE
Render page break label as an UI element

### LINKED ISSUE #8788: Dropping image over the page break crashes the editor
## 📝 Provide detailed reproduction steps (if any)

1.  https://ckeditor.com/docs/ckeditor5/latest/features/page-break.html
2.  Drag and drop some image over the page break widget

### ✔️ Expected result

An image should be properly dropped to the editor.

### ❌ Actual result

```
Uncaught CKEditorError: model-nodelist-offset-out-of-bounds {"offset":9,"nodeList":[]}
Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-model-nodelist-offset-out-of-bounds
    at offsetToIndex (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:141138)
    at c.offsetToIndex (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:44806)
    at a (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:15894)
    at r.get textNode [as textNode] (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:10039)
    at $e (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:426548)
    at qe._validateSelectionRange (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:426384)
    at m.<anonymous> (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:94898)
    at m.fire (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:28590)
    at m._setRanges (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:143980)
    at m.setTo (https://ckeditor.com/docs/ckeditor5/latest/snippets/features/page-break/snippet.js:4:143626)
```

## 📃 Other details

*   Browser: …
*   OS: …
*   First affected CKEditor version: …
*   Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by maxbarnas**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (page-break): Drag and dropping image on Page Break divider is no longer crashing the editor. Closes #8788.

⚠️ Merge this, then related PRs! ⚠️

### CODE REVIEW

File: packages/ckeditor5-page-break/src/pagebreakediting.js [hunk 1]
```diff
@@ -66,16 +66,19 @@ export default class PageBreakEditing extends Plugin {
...
+					function( domDocument ) {
+						const domElement = this.toDomElement( domDocument );
+						domElement.innerText = t( 'Page break' );
+
+						return domElement;
+					}
+				);
 
 				writer.addClass( 'page-break', viewWrapper );
-				writer.setCustomProperty( 'pageBreak', true, viewWrapper );
```
Comment (niegowski) [inline packages/ckeditor5-page-break/src/pagebreakediting.js:73]: This should be preserved in the new code.

File: packages/ckeditor5-page-break/tests/pagebreakediting.js [hunk 2]
```diff
@@ -252,7 +252,7 @@ describe( 'PageBreakEditing', () => {
 			it( 'should convert', () => {
 				setModelData( model, '<pageBreak></pageBreak>' );
 
-				expect( getViewData( view, { withoutSelection: true } ) ).to.equal(
+				expect( getViewData( view, { withoutSelection: true, renderUIElements: true } ) ).to.equal(
```
Comment (niegowski) [inline packages/ckeditor5-page-break/tests/pagebreakediting.js:255]: I would add this as a separate check (could be inside the same test) so one with `renderUIElements` and another one without it.

**Review by niegowski — CHANGES_REQUESTED** [review]
As commented below.
Also, it would be good to have a test for the original issue.

File: packages/ckeditor5-page-break/src/pagebreakediting.js (hunk 1 above)
Comment (maxbarnas) [reply packages/ckeditor5-page-break/src/pagebreakediting.js:73]: It's alrady implemented by `toPageBreakWidget()`: https://github.com/ckeditor/ckeditor5/blob/b1d465c/packages/ckeditor5-page-break/src/pagebreakediting.js#L141.

Comment (niegowski) [reply packages/ckeditor5-page-break/src/pagebreakediting.js:73]: Oh, I missed it. So it was updated twice in the current version

Comment (maxbarnas) [reply packages/ckeditor5-page-break/src/pagebreakediting.js:73]: Haha, yup. I was so impressed with this "optimisation" that I forgot to test the original case. Thanks for pointing that out! :D

File: packages/ckeditor5-page-break/tests/pagebreakediting.js [hunk 3]
```diff
@@ -50,6 +50,16 @@ describe( 'PageBreakEditing', () => {
...
 	} );
 
+	// https://github.com/ckeditor/ckeditor5/issues/8788.
+	// Proper integration testing of this is too complex.
+	// Making sure the label is no longer a regular text element should be enough.
+	it( 'should have label as a UIElement', () => {
+		setModelData( model, '[<pageBreak></pageBreak>]' );
+		const textNode = viewDocument.getRoot().getChild( 0 ).getChild( 0 );
+
+		expect( textNode.is( 'uiElement' ) ).to.be.true;
```
Comment (niegowski) [inline packages/ckeditor5-page-break/tests/pagebreakediting.js:60]: There could be more `UIElements` inside this widget (i.e., Widget Type Around feature) so we should be more precise in this test.

**Pull request merged**